src/scoped.rs

```diff
@@ -146,47 +144,58 @@ pub unsafe fn builder_spawn_unsafe<'a, F>(
 where
     F: FnOnce() + Send + 'a,
 {
-    use std::mem;
-
-    let closure: Box<FnBox + 'a> = Box::new(f);
-    let closure: Box<FnBox + Send> = mem::transmute(closure);
+    let closure: Box<FnBox<()> + 'a> = Box::new(f);
+    let closure: Box<FnBox<()> + Send> = mem::transmute(closure);
     builder.spawn(move || closure.call_box())
 }
 
-
 pub struct Scope<'a> {
-    dtors: RefCell<Option<DtorChain<'a>>>,
+    dtors: RefCell<Option<DtorChain<'a, ()>>>,
+    joins: RefCell<Option<DtorChain<'a, Option<thread::Result<()>>>>>,
```

Comment:
What is the reason for this being an `Option<thread::Result<()>>`? It seems that every joined thread has a result, so maybe just `thread::Result<()>` would work.

Reply:
I explained it in a new commit: "It contains `Option<thread::Result<()>>` (instead of `thread::Result<()>`) because `ScopedJoinHandle` may have already called `join()` and used the join result."

Reply:
I see, so the problem seems to be that `ScopedJoinHandle`s don't pop their `JoinState`s off the `DtorChain` like `Scope::join` does, but rather leave a `None` in their place. Maybe this is fine.